src/ScanItems.js

```diff
@@ -308,26 +317,38 @@ class ScanItems extends React.Component {
     return data;
   }
 
+  getCheckoutMutator = () => {
+    const {
+      stripes,
+      mutator: {
+        checkout,
+        checkoutBFF,
+      },
+    } = this.props;
+    const isEnabledEcsRequests = stripes?.config?.enableEcsRequests;
+
+    return isEnabledEcsRequests ? checkoutBFF : checkout;
```

Comment:
Please include a `hasInterface` check: config and actual installed modules would both have to be in alignment.

```
return isEnabledEcsRequests && hasInterface('circulation-bff-loans', '1.0') ? checkoutBFF : checkout;
```

Reply:
Hello @JohnC-80 
We want to be sure that regardless of availability of `circulation-bff-loans` interface we will use `circulation-bff/loans/check-out-by-barcode` endpoint.
At this stage this is acceptable to us. And we do not need to trigger `circulation/check-out-by-barcode` endpoint if `circulation-bff-loans` interface does not exist by some reason. It can lead to some potential issues.

cc @Dmitriy-Litvinenko

Reply:
If someone configures `isEnabledEcsRequests : true ` without having `circulation-bff-loans` module installed , the endpoint would not exist, is that correct? Checking the availability of the interface is the guard against this situation.

Reply:
Yes, it is correct. And in this case we are expecting to trigger not existed endpoint with failed response instead of sending data to old endpoint. 
If in ECS env we send data to old endpoint we will have incorrect data handling on the server side. It can lead to potential issues.
Currently work related to bff and not bff endpoints is in progress. A bit later we will use only bff endpoints and functionality that depends on `isEnabledEcsRequests` environment variable will be deleted.

Reply:
> If someone configures `isEnabledEcsRequests : true ` without having `circulation-bff-loans` module installed , the endpoint would not exist, is that correct?
Hello @JohnC-80 Yes, it is correct. When we have `isEnabledEcsRequests: true` we must use `circulation-bff/loans/check-out-by-barcod`. When we do not have `isEnabledEcsRequests: true` we should not use `circulation-bff/loans/check-out-by-barcod`. All other cases this is configuration issue and we should found them as soon as possible and fix them. We should not try to work with not corresponding endpoint, because our application will work incorrect and we will have problems/errors/bugs and etc.